controller/src/main/java/io/pravega/controller/store/stream/tables/serializers/RetentionRecordSerializerV1.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream.tables.serializers;
+
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.controller.store.stream.StreamCutRecord;
+import io.pravega.controller.store.stream.tables.RetentionRecord;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class RetentionRecordSerializerV1 extends VersionedSerializer.WithBuilder<RetentionRecord, RetentionRecord.RetentionRecordBuilder> {
```

Comment:
One thing we can do, in order to keep the number of files small, is to make this a nested class of `RetentionRecord` and only expose it via a singleton `public static final` field (just like you do).